### PR TITLE
[autograd] Route two-sided clamp.Tensor backward through a private

### PR DESCRIPTION
**Pull request by PatrykWilczewski**
dispatched op

- Problem: pytorch/pytorch#191142 rewrote the two-sided clamp.backward(gard, self, Tensor min, Tensor max) self-gradient from 5 to 11 element-wise ops in order to split the 1/2-gradient correctly at ties. The chain is inlined in FunctionsManual.cpp, so no backend can fuse it and every caller pays 11 kernel dispatches. On launch-bound hardware this is expensive: hf_T5_large float16 training issues 120 such calls per training step, and on an Arc B580 the 11-op chain costs roughly 380% more per call than the 5-op version it replaced.
- Root cause: the subgradient formula is hard-coded inside the autograd function. There is no dispatch point at which a backend can provide a fused kernel, and no ATen op that tracers can reason about or decompose.
- Fix: extract the two-sided path into a new private op aten::_clamp_backward_tensor.
  - The CompositeExplicitAutograd kernel (_clamp_backward_tensor_composite in native/TensorCompare.cpp) is the existing 11-op sequence moved verbatim, so behavior is unchanged on every backend.
  - clamp_backward() in FunctionsManual.cpp now calls the op. The min-only and amx-only branches are left untouched.
  - derivatives.yaml entry: the mask depends on self/min/max only through comparisons, so those carry no second-order gradient, and grad_output is re-masked exactly as in the first order.
  - A decomposition is registered in torch/_decomp so tracers don't see an opaque node. This is required: CompositeExplicitAutograd ops are NOT auto-decomposed durning AOTAutograd tracing, so without it the new op becomes an inductor extern-kernel fallback and the 11 element-wise ops stop fusing into neighbouring backward kernels on every backend. Measured on an Arc B580 at hf_T5_large's real shape ([2,128,1024] float16 activations, 0-dim float32 bounds): without the decomposition inductor emits one extern-kernel fakkback and 4 triton kernels; with it, no fallback and 3 triton kernels, and the compiled backward is 14% faster. No XPU dispatch entry is added here. It follows in a separate change once the fused kernel exists in torch-xpu-ops, because the kernel needs this op's generated header to compile while the dispatch entry needs the kernel's symbol to link.
- Validation:
  - pytest test/test_binary_ufuncs.py -k "clamp and xpu": 486 passed
  - pytest test/test_ops.py -k "clamp and xpu": 90 passed
  - pytest test/test_meta.py -k clamp: 282 passed
  - pytest test/test_decomp.py -k clamp: 58 passed
  - PYTORCH_TEST_WITH_SLOW=1 pytest test/test_schema_check.py -k clamp: 29 passed
  - pytest test/test_decomp.py -k "TestDecompOpInfo or table or opset or core": 50 passed
  - pytest test/export/test_export.py -k decomp: 2 passed
  - the decomposition was verified bit-exact against the composite across boundary cases (equal, reversed and reversed-at-min bounds, NaN self), all four float dtypes, mixed and reverse-mixed bound dtypes, broadcast and non-contiguous inputs, and a 4096 randomized elements drawn so that exact ties are hit; gradcheck and gradgradcheck pass.
- Additional tests: added - a mixed-dtype clamp subgradient test.

Related: intel/torch-xpu-ops#5155

### CODE REVIEW

**Review by sylvesterkaczmarek — COMMENTED** [review]
Introducing a private dispatch point is a sensible way to recover backend fusion without changing the existing subgradient formula. The decomposition is load-bearing here: without it, AOTAutograd/Inductor would see an opaque CompositeExplicitAutograd op and could lose fusion on backends that do not provide a native kernel. Keeping the tensor-bound comparisons in their original dtypes also preserves the mixed-bound semantics covered by the XPU regression.
